dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -306,24 +305,29 @@ def diff(self, other_profile: CategoricalColumn, options: dict = None) -> dict:
                     other_profile._categories.items(), key=itemgetter(1), reverse=True
                 )
             )
-            if cat_count1.keys() == cat_count2.keys():
-                total_psi = 0.0
-                for key in cat_count1.keys():
-                    perc_A = cat_count1[key] / self.sample_size
-                    perc_B = cat_count2[key] / other_profile.sample_size
-                    total_psi += (perc_B - perc_A) * math.log(perc_B / perc_A)
-                    differences["statistics"]["psi"] = total_psi
-            else:
-                warnings.warn(
-                    "psi was not calculated due to the differences in categories "
-                    "of the profiles. Differences:\n"
-                    f"{set(cat_count1.keys()) ^ set(cat_count2.keys())}",
-                    RuntimeWarning,
-                )
+            (
+                self_cat_count,
+                other_cat_count,
+            ) = self._preprocess_for_categorical_psi_calculation(
+                self_cat_count=cat_count1,
+                other_cat_count=cat_count2,
+            )
+
+            total_psi = 0.0
+            for iter_key in self_cat_count.keys():
+                percent_self = self_cat_count[iter_key] / self.sample_size
+                percent_other = other_cat_count[iter_key] / other_profile.sample_size
+                if (percent_other == 0) or (percent_self == 0):
+                    total_psi += 0.0
```

Comment:
this `total_psi += 0.0` isn't really needed but added for readability and clarity on what is occuring